### PR TITLE
Center sprites and text in adventure map treasure chest dialog

### LINKED ISSUE #9025: Adventure Map, Treasure Chest window, sprites for the gold pile and experience points star, and the text below the sprites, are not centered horizontally in relation to the position of the corresponding "YES" "NO" buttons
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

In fheroes2, the sprites for the gold pile and experience points star, and the text below the sprites, are not centered horizontally in relation to the position of the corresponding "YES" "NO" buttons.

In fheroes2:

XP Need correction 01

**Edit 2024/08/05.**

If necessary, the text can be arranged on several lines, for example:

"500
(XP Needed: 1000)"

Or:

"500

XP Needed:

1000"

With:

"1000
(Total: 6000)"

Or:

"1000

Total:

6000"

Or even, the size of the window can be enlarged horizontally.

### Save file

Not needed.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by IvanAtanascev**
Fixes #9025 by centering the gold pile and XP sprites horizontally in relation to their respective buttons ("YES" and "NO") in the adventure map treasure chest dialog window.

### CODE REVIEW

File: src/fheroes2/dialog/dialog_chest.cpp [hunk 1]
```diff
@@ -71,10 +71,14 @@ bool Dialog::SelectGoldOrExp( const std::string & header, const std::string & me
...
     fheroes2::Button button_yes( pt.x, pt.y, buttonYesIcnID, 0, 1 );
 
+    const int buttonYesXPosition = pt.x;
+
     pt.x = box.GetArea().x + box.GetArea().width / 2 + 20;
     pt.y = box.GetArea().y + box.GetArea().height - fheroes2::AGG::GetICN( buttonNoIcnID, 0 ).height();
     fheroes2::Button button_no( pt.x, pt.y, buttonNoIcnID, 0, 1 );
 
+    const int buttonNoXPosition = pt.x;
+
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_chest.cpp:81]: We can get rid of `pt` (line 64) and cache the `fheroes2::AGG::GetICN( buttonYesIcnID, 0 )`:
```cpp
const fheroes2::Sprite & buttonYesSprite = fheroes2::AGG::GetICN( buttonYesIcnID, 0 );

const int32_t buttonsYPosition = box.GetArea().y + box.GetArea().height - buttonYesSprite.height();
const int32_t buttonYesXPosition = box.GetArea().x + box.GetArea().width / 2 - buttonYesSprite.width() - 20;

fheroes2::Button button_yes( buttonYesXPosition , buttonsYPosition, buttonYesIcnID, 0, 1 );

const int32_t buttonNoXPosition = box.GetArea().x + box.GetArea().width / 2 + 20;

fheroes2::Button button_no( buttonNoXPosition , buttonsYPosition, buttonNoIcnID, 0, 1 );
```
Buttons' height is the same so we can use one `buttonsYPosition` for the both buttons.

File: src/fheroes2/dialog/dialog_chest.cpp [hunk 2]
```diff
@@ -91,13 +95,13 @@ bool Dialog::SelectGoldOrExp( const std::string & header, const std::string & me
 
     pos.y += sprite_expr.height();
     // sprite1
-    pos.x = box.GetArea().x + box.GetArea().width / 2 - sprite_gold.width() - 30;
+    pos.x = buttonYesXPosition + ( ( fheroes2::AGG::GetICN( buttonYesIcnID, 0 ).width() - sprite_gold.width() ) / 2 );
```
Comment (Districh-ru) [inline src/fheroes2/dialog/dialog_chest.cpp:98]: Then we can use `buttonYesSprite` here too:
`pos.x = buttonYesXPosition + ( ( buttonYesSprite.width() - sprite_gold.width() ) / 2 );`
:)

**Review by Districh-ru — CHANGES_REQUESTED** [review]
Hi, @IvanAtanascev, the fix looks good to me:
изображение

I left two suggestions to clean-up the changed code. Could you please check them when you have time.

**Comment by IvanAtanascev** [comment]
Hello, @Districh-ru, thank you for the suggestions. I have applied them in 495bb58. The code looks better this way and works as expected.

**Review by Districh-ru — APPROVED** [review]

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@IvanAtanascev , thank you so much for your contribution to the project!

**Pull request merged**